core/src/main/scala/cats/Traverse.scala

```diff
@@ -30,6 +30,12 @@ import simulacrum.typeclass
     U.TC.traverse(fa)(a => U.subst(f(a)))(this)
 
   /**
+   * A traverse followed by flattening the inner result.
+   */
+  def traverseM[G[_], A, B](fa: F[A])(f: A => G[F[B]])(implicit G: Applicative[G], F: FlatMap[F]): G[F[B]] =
+    G.map(traverse(fa)(f))(F.flatten)
```

Comment:
I think the version that you currently have is probably better, because `Applicative.traverse` ends up delegating through to `Traverse.traverse` anyway.